sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GlobalEndpointManager.java

```diff
@@ -310,6 +323,7 @@ private Mono<DatabaseAccount> getDatabaseAccountAsync(URI serviceEndpoint) {
             .doOnNext(databaseAccount -> {
                 if(databaseAccount != null) {
                     this.latestDatabaseAccount = databaseAccount;
+                    this.setLatestDatabaseRefreshError(null);
```

Comment:
Setting this to null might reflect as `NullPointerException` somewhere, IMO, it would be better to set it to empty string.

Reply:
Why? I think as ong as it is consumed correctly it shoudl be fine. Initial version was indeed not thread-safe - I have changed that part